parsl/providers/ad_hoc/ad_hoc.py

```diff
@@ -0,0 +1,267 @@
+        if channel is None:
+            logger.warning("All Channels in Ad-Hoc provider are in use")
+            return None
+
+        job_name = "{0}.{1}".format(job_name, time.time())
+
+        # Set script path
+        script_path = "{0}/{1}.sh".format(self.script_dir, job_name)
+        script_path = os.path.abspath(script_path)
+
+        wrap_command = self.worker_init + '\n' + self.launcher(command, tasks_per_node, self.nodes_per_block)
+
+        self._write_submit_script(wrap_command, script_path)
+
+        job_id = None
+        proc = None
+        remote_pid = None
+        final_cmd = None
+
+        if (self.move_files is None and not isinstance(channel, LocalChannel)) or (self.move_files):
+            logger.debug("Pushing start script")
+            script_path = channel.push_file(script_path, channel.script_dir)
+
+        if not isinstance(channel, LocalChannel):
+            # Bash would return until the streams are closed. So we redirect to a outs file
+            final_cmd = 'bash {0} &> {0}.out & \n echo "PID:$!" '.format(script_path)
+            retcode, stdout, stderr = channel.execute_wait(final_cmd, self.cmd_timeout)
+            for line in stdout.split('\n'):
+                if line.startswith("PID:"):
+                    remote_pid = line.split("PID:")[1].strip()
+                    job_id = remote_pid
+            if job_id is None:
+                logger.warning("Channel failed to start remote command/retrieve PID")
+        else:
+            try:
+                final_cmd = 'bash {0}'.format(script_path)
+                job_id, proc = channel.execute_no_wait(final_cmd, self.cmd_timeout)
+            except Exception as e:
+                logger.debug("Channel execute failed for: {}, {}".format(channel, e))
+                raise
+
+        self.resources[job_id] = {'job_id': job_id,
+                                  'status': 'RUNNING',
+                                  'cmd': final_cmd,
+                                  'channel': channel,
+                                  'remote_pid': remote_pid,
+                                  'proc': proc}
+
+        return job_id
+
+    def status(self, job_ids):
+        """ Get status of the list of jobs with job_ids
+
+        Parameters
+        ----------
+        job_ids : list of strings
+          List of job id strings
+
+        Returns
+        -------
+        list of status strings ['PENDING', 'COMPLETED', 'FAILED']
+        """
+        for job_id in job_ids:
+            channel = self.resources[job_id]['channel']
+            status_command = "ps --pid {} | grep {}".format(self.resources[job_id]['job_id'],
+                                                            self.resources[job_id]['cmd'].split()[0])
+            retcode, stdout, stderr = channel.execute_wait(status_command)
+            if retcode != 0 and self.resources[job_id]['status'] == 'RUNNING':
+                self.resources[job_id]['status'] == 'FAILED'
```

Comment:
This should be `=` not `==`. With `==`, we never notice that a block has died, and scaling up to fill in broken blocks doesn't happen.

Reply:
There were two `=` vs `==` bugs. Fixed.